Phalanx: turn PHX_ALLOW_MULTIPLE_EVALUATORS_FOR_SAME_FIELD off by default

<!---
Provide a general summary of the issue in the Title above.  If this issue
pertains to a particular package in Trilinos, it's worthwhile to start the
title with "PackageName:  ".
-->

<!---
Note that anything between these delimiters is a comment that will not appear
in the issue description once created.  Click on the Preview tab to see what
everything will look like when you submit.
-->

<!---
Feel free to delete anything from this template that is not applicable to the
issue you are submitting.
-->

<!---
Replace <teamName> below with the appropriate Trilinos package/team name.
-->
@trilinos/Phalanx

<!---
Assignees:  If you know anyone who should likely tackle this issue, select them
from the Assignees drop-down on the right.
-->

<!---
Lables:  Choose any applicable package names from the Labels drop-down on the
right.  Additionally, choose a label to indicate the type of issue, for
instance, bug, build, documentation, enhancement, etc.
-->

## Expectations
<!---
Tell us what you think should happen, how you think things should work, what
you would like to see in the documentation, etc.
-->
I _think_ more likely than not one wants to see an error if two evaluators evaluate the same field. I am ok with keeping the feature (CMakeLists.txt explain it is for backward compatibility), but should not be the default. It is more costly to debug an app that wrongly has two evaluators evaluating the same field, than it is for another app to add the configure option to allow this. The former, can take hours of debugging (I did this mistake _twice_); the latter takes one failed run (with meaningful error message) to adjust the installation.

## Current Behavior
<!---
Tell us how the current behavior fails to meet your expectations in some way.
-->
The flag is on by default, so PHX allows multiple evaluators to evaluate the same field, in which case, the first is used, and the others tossed. If one does _not_ want this feature, the option must be explicitly turned off at configure time.

Note: this option was introduced in 2015, and left ON by default for bwd compatibility in existing codes. I believe 3 years is a reasonable time for codes to either adapt, or add a config option to their trilinos configuration.

## Motivation and Context
<!---
How has this expectation failure affected you?  What are you trying to
accomplish?  Why do we need to address this?  What does it have to do with
anything?  Providing context helps us come up with a solution that is most
useful in the real world.
-->
It is not unlikely to fall in the scenario where 2 evaluators evaluate the same field. Perhaps you used to have `OldEvaluator` to compute `F`, but now you wrote `VeryCoolNewEvaluator` to evaluate `F`. So you add it to your problem, but forget to delete the previous one. Or worse yet: two evaluators _happen_ to evaluate the same field because of a particular clash in your app parameters, in which case, you probably need to rethink your app flow (so you would _love_ to see an error message).

## Possible Solution
<!---
Not obligatory, but suggest a fix for the bug or documentation, or suggest
ideas on how to implement the addition or change.
-->
Simply change the default to OFF. Or remove the option altogether. It may be that it was meant to achieve the same thing that _contributed_ fields now allow. So _perhaps_ this is no longer needed.

## Steps to Reproduce
<!---
Provide a link to a live example, or an unambiguous set of steps to reproduce
this issue.  Include code to reproduce, if relevant. 
1. Do this.
1. Do that.
1. Shake fist angrily at computer.
-->
Take your favorite problem, and register the same evaluator twice, and see if you get an error. Yes, in that case everything would be fine, since the two evaluators are the same, but it still exposes the problem: no errors with the default PHX configuration.

@bartlettroscoe - If I try to use that I get a configure error. Looks like the flag is empty. What does an empty variable imply - on/off/unknown?

```
CMake Error at packages/phalanx/CMakeLists.txt:15 (TRIBITS_ADD_OPTION_AND_DEFINE):
  TRIBITS_ADD_OPTION_AND_DEFINE Macro invoked with incorrect arguments for
  macro named: TRIBITS_ADD_OPTION_AND_DEFINE


-- Phalanx_ALLOW_MULTIPLE_EVALUATORS_FOR_SAME_FIELD = 
```

--- a/packages/phalanx/CMakeLists.txt
+++ b/packages/phalanx/CMakeLists.txt
@@ -15,7 +15,7 @@ TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
 TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ALLOW_MULTIPLE_EVALUATORS_FOR_SAME_FIELD
   PHX_ALLOW_MULTIPLE_EVALUATORS_FOR_SAME_FIELD
   "Enable to use backwards compatible mode where multiple evaluators can be registered that evalaute the same field. Phalanx will use the first evaluator it finds. Default to OFF but unless Panzer is enabled. For backwards compatibility with Panzer users, we default this to ON."
-  ${PROJECT_NAME}_ENABLE_Panzer )
+  "${${PROJECT_NAME}_ENABLE_Panzer}" )
 MESSAGE(STATUS "Allow Multiple Evaluator Registration for Same Field: ${${PACKAGE_NAME}_ALLOW_MULTIPLE_EVALUATORS_FOR_SAME_FIELD}")
 
 TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_EXPLICIT_TEMPLATE_INSTANTIATION

--- a/packages/phalanx/CMakeLists.txt
+++ b/packages/phalanx/CMakeLists.txt
@@ -14,12 +14,13 @@ TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
 
 TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ALLOW_MULTIPLE_EVALUATORS_FOR_SAME_FIELD
   PHX_ALLOW_MULTIPLE_EVALUATORS_FOR_SAME_FIELD
-  "Enable to use backwards compatible mode where multiple evaluators can be registered that evalaute the same field. Phalanx will use the first evaluator it finds. This should default to off but for backwards compatibility with current codes, we default this to ON."
-  ON )
+  "Enable to use backwards compatible mode where multiple evaluators can be registered that evalaute the same field. Phalanx will use the first evaluator it finds. Default to OFF but unless Panzer is enabled. For backwards compatibility with Panzer users, we default this to ON."
+  ${PROJECT_NAME}_ENABLE_Panzer )
+MESSAGE(STATUS "Allow Multiple Evaluator Registration for Same Field: ${${PACKAGE_NAME}_ALLOW_MULTIPLE_EVALUATORS_FOR_SAME_FIELD}")
 
 TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_EXPLICIT_TEMPLATE_INSTANTIATION
   PHX_ETI
-  "Build with explicit template instation enabled."
+  "Build with explicit template instation enabled in the examples."
   OFF )
 
 # If not explicitly defined by the user, pick a default node type that
@@ -98,6 +99,9 @@ TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ENABLE_TEUCHOS_TIME_MONITOR
   "Build with Teuchos TimeMonitors wrapped around each evaluator."
   "${${PROJECT_NAME}_ENABLE_TEUCHOS_TIME_MONITOR}" )
 
+# Use only for CUDA builds
+MESSAGE(STATUS "Sacado Hierarchic DFAD support: ${Sacado_ENABLE_HIERARCHICAL_DFAD}")
+
 # Experimental device dag support
 TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ENABLE_DEVICE_DAG
   PHX_ENABLE_DEVICE_DAG
@@ -124,8 +128,6 @@ IF(${PACKAGE_NAME}_ENABLE_KOKKOS_AMT)
   MESSAGE(FATAL_ERROR "Phalanx configure ERROR: Kokkos AMT support is enabled but is not implemented for Pthreads execution space. Please change your execution space or disable AMT support in Phalanx.")
 ENDIF()
 ENDIF(${PACKAGE_NAME}_ENABLE_KOKKOS_AMT)
-
-MESSAGE(STATUS "Sacado Hierarchic DFAD support: ${Sacado_ENABLE_HIERARCHICAL_DFAD}")
 
 # For use by Phalanx and Panzer unit tests
 GLOBAL_SET(PHALANX_UNIT_TEST_MAIN

